File: Phase01/src/SearchQuery.java
import java.util.HashSet;

public class SearchQuery {

    private HashSet<String> mustIncludeWords;
    private HashSet<String> includeWords;
    private HashSet<String> excludeWords;

    private SearchQuery() {
        mustIncludeWords = new HashSet<>();
        includeWords = new HashSet<>();
        excludeWords = new HashSet<>();
    }

    public HashSet<String> getMustIncludeWords() {
        return mustIncludeWords;
    }

    public HashSet<String> getIncludeWords() {
        return includeWords;
    }

    public HashSet<String> getExcludeWords() {
        return excludeWords;
    }

    public static SearchQuery parseString(String query) {
        SearchQuery searchQuery = new SearchQuery();
        String[] subs = query.split(" ");
        addSingleWord(searchQuery, subs);
        return searchQuery;
    }

    private static void addSingleWord(SearchQuery searchQuery, String[] subs){
        for (String word : subs) {
            switch(word.charAt(0)){
                case '+':
                     searchQuery.includeWords.add(word.substring(1));
                     break;
                 case '-':
                     searchQuery.excludeWords.add(word.substring(1));
                     break;
                 default: 
                     searchQuery.mustIncludeWords.add(word);
             }
        }
    }
}

File: C#/Phase08/InvertedIndexEngine/Model/SearchQuery.cs
using System.Collections.Generic;

namespace InvertedIndexEngine.Model
{
    public class SearchQuery
    {

        public readonly HashSet<string> MustIncludeWords;
        public readonly HashSet<string> IncludeWords;
        public readonly HashSet<string> ExcludeWords;

        public SearchQuery(string query)
        {
            MustIncludeWords = new HashSet<string>();
            IncludeWords = new HashSet<string>();
            ExcludeWords = new HashSet<string>();
            this.ParseString(query);
        }


        public void ParseString(string query)
        {
            var words = query.Trim().Split(' ');
            if (words[0].Length > 0)
            {
                foreach (var word in words)
                {
                    AddToCorrectSet(word);
                }
            }
        }

        private void AddToCorrectSet(string word)
        {
            switch (word.Substring(0, 1))
            {
                case "+":
                    IncludeWords.Add(word);
                    break;
                case "-":
                    ExcludeWords.Add(word);
                    break;
                default:
                    MustIncludeWords.Add(word);
                    break;
            }
        }
    }
}